src/backend/aspen/api/conftest.py

```diff
@@ -0,0 +1,122 @@
+import random
+import string
+from dataclasses import dataclass
+from functools import partial
+from typing import AsyncGenerator
+
+import pytest
+from fastapi import Depends, FastAPI, Request
+from httpx import AsyncClient
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from aspen.api.auth import get_auth_user, setup_userinfo
+from aspen.api.deps import get_db
+from aspen.api.main import get_app
+from aspen.database import connection as aspen_connection
+from aspen.database import schema
+from aspen.database.connection import init_async_db
+
+USERNAME = "user_rw"
+PASSWORD = "password_rw"
+INITIAL_DATABASE = "postgres"
+
+pytestmark = pytest.mark.anyio
+
+
+@dataclass
+class AsyncPostgresDatabase:
+    database_name: str
+    port: int
+
+    def as_uri(self) -> str:
+        return f"postgresql+asyncpg://{USERNAME}:{PASSWORD}@database:{self.port}/{self.database_name}"
+
+
+@pytest.fixture()
+async def async_db() -> AsyncGenerator[AsyncPostgresDatabase, None]:
+    """Creates a postgres test database named a random string with username/password user_rw/password_rw, yields it, and then drops it."""
+
+    # create admin sql connection to create test database
+    admin_uri = "postgresql+asyncpg://postgres:password_postgres@database:5432/aspen_db"
```

Comment:
This feels like something that should be using env vars rather than hardcoded (not that it's actually likely to change anytime soon).